sui_programmability/adapter/src/adapter.rs

```diff
@@ -207,11 +181,13 @@ fn execute_internal<
             let updated_ctx: TxContext = bcs::from_bytes(&ctx_bytes).unwrap();
             ctx.update_state(updated_ctx)?;
 
-            let mutable_refs = mutable_ref_objects.into_iter().zip(
+            let mutable_refs =
                 mutable_reference_outputs
                     .into_iter()
-                    .map(|(_local_idx, bytes, _layout)| bytes),
-            );
+                    .map(|(local_idx, bytes, _layout)| {
+                        let object = mutable_ref_objects.remove(&(local_idx as usize)).unwrap();
+                        (object, bytes)
+                    });
```

Comment:
For clarity, might want to add assert that mutable_ref_objects will be empty at the end